serde_derive_internals/src/ast.rs

```diff
@@ -98,44 +98,61 @@ impl<'a> Body<'a> {
     }
 }
 
-fn enum_from_ast<'a>(cx: &Ctxt, variants: &'a [syn::Variant]) -> Vec<Variant<'a>> {
+fn enum_from_ast<'a>(
+    cx: &Ctxt,
+    variants: &'a [syn::Variant],
+    container: &attr::Container,
```

Comment:
I wonder if we should just pass a reference to the `default` field. Information frugality and stuff ^^

Reply:
It's a trade-off between the `Container` knowing that `Field` does inside, and `Field` knowing what `Container` does inside. I decided on the latter, but switching it around seems fine. I'm on it.

Reply:
hm? I meant just passing `container_default: &Default`